metadata-ingestion/src/datahub/ingestion/source/kafka.py

```diff
@@ -298,15 +298,16 @@ def _extract_record(self, topic: str) -> Iterable[MetadataWorkUnit]:
         dataset_snapshot.aspects.append(browse_path)
 
         # 4. Attach dataPlatformInstance aspect.
-        if self.source_config.platform_instance:
-            dataset_snapshot.aspects.append(
-                DataPlatformInstanceClass(
-                    platform=platform_urn,
-                    instance=make_dataplatform_instance_urn(
-                        self.platform, self.source_config.platform_instance
-                    ),
+        dataset_snapshot.aspects.append(
```

Comment:
We need to rethink what is the invariant we want when we find a data-platform-instance aspect attached to an entity. Is it that the recipe author provided a specific platform_instance to use? or that the ingestion code auto-generated an instance based on the ENV. 

The dataplatform instance migration script depends on finding no data platform instance aspect or finding a data platform instance without the instance specified, to figure out which entities to migrate... so it might get trickier if we have lots of entities sitting around with "default instances" being populated. 

We shouldn't accept this PR until we have figured out what the assertion is.